From 1f774a5a92dde35388a0afb71554925682b01428 Mon Sep 17 00:00:00 2001
From: Yannick FERTRE <yannick.fertre@foss.st.com>
Date: Fri, 25 Aug 2023 10:21:08 +0200
Subject: [PATCH 0449/1141] drm/stm: dsi: Don't check data timings

Data timings hs2lp & lp2hs don't have to be check
(DSI hardware version 1.41 only) to allows to support
resolution 1080P60 (threw the hdmi bridge adv7535).

Change-Id: I3e1b5a43d549ca952fad637537d5d23fd3d4a06a
Signed-off-by: Yannick FERTRE <yannick.fertre@foss.st.com>
---
 drivers/gpu/drm/stm/dw_mipi_dsi-stm.c | 27 +++++++++++++--------------
 1 file changed, 13 insertions(+), 14 deletions(-)

diff --git a/drivers/gpu/drm/stm/dw_mipi_dsi-stm.c b/drivers/gpu/drm/stm/dw_mipi_dsi-stm.c
index 845d27b772a1..c95f3d445881 100644
--- a/drivers/gpu/drm/stm/dw_mipi_dsi-stm.c
+++ b/drivers/gpu/drm/stm/dw_mipi_dsi-stm.c
@@ -841,21 +841,20 @@ dw_mipi_dsi_stm_mode_valid(void *priv_data,
 
 		lane_mbps = pll_out_khz / 1000;
 
-		if (dsi->hw_version == HWVER_141)
-			ret = dw_mipi_dsi_phy_141_get_timing(priv_data, lane_mbps, &dphy_timing);
-		else
+		if (dsi->hw_version < HWVER_141) {
 			ret = dw_mipi_dsi_phy_get_timing(priv_data, lane_mbps, &dphy_timing);
-		if (ret)
-			return MODE_ERROR;
-		/*
-		 * In non-burst mode DSI has to enter in LP during HFP
-		 * (horizontal front porch) or HBP (horizontal back porch) to
-		 * resync with LTDC pixel clock.
-		 */
-		delay_to_lp = DIV_ROUND_UP((dphy_timing.data_hs2lp + dphy_timing.data_lp2hs) *
-					   lanes * BITS_PER_BYTE, bpp);
-		if (hfp < delay_to_lp && hbp < delay_to_lp)
-			return MODE_HSYNC;
+			if (ret)
+				return MODE_ERROR;
+			/*
+			 * In non-burst mode DSI has to enter in LP during HFP
+			 * (horizontal front porch) or HBP (horizontal back porch) to
+			 * resync with LTDC pixel clock.
+			 */
+			delay_to_lp = DIV_ROUND_UP((dphy_timing.data_hs2lp + dphy_timing.data_lp2hs)
+						   * lanes * BITS_PER_BYTE, bpp);
+			if (hfp < delay_to_lp && hbp < delay_to_lp)
+				return MODE_HSYNC;
+		}
 	}
 
 	return MODE_OK;
-- 
2.39.2

